pkg/gardenlet/operation/botanist/infrastructure.go

```diff
@@ -63,13 +63,15 @@ func (b *Botanist) WaitForInfrastructure(ctx context.Context) error {
 	}
 
 	networkingStatus := &gardencorev1beta1.NetworkingStatus{}
-
 	if nodesCIDRs := b.Shoot.Components.Extensions.Infrastructure.NodesCIDRs(); len(nodesCIDRs) > 0 {
-		if err := b.Shoot.UpdateInfo(ctx, b.GardenClient, true, func(shoot *gardencorev1beta1.Shoot) error {
-			shoot.Spec.Networking.Nodes = &nodesCIDRs[0]
-			return nil
-		}); err != nil {
-			return err
+		// Only update node CIDR in IPv6 single stack clusters.
+		if b.Shoot.GetInfo().Spec.Networking.Nodes == nil {
+			if err := b.Shoot.UpdateInfo(ctx, b.GardenClient, true, func(shoot *gardencorev1beta1.Shoot) error {
+				shoot.Spec.Networking.Nodes = &nodesCIDRs[0]
+				return nil
+			}); err != nil {
+				return err
+			}
 		}
```

Comment:
Should this comment not rather read that we only update `spec.networking.nodes` if not set, yet? It can happen in the IPv6-only single-stack case, but also in other scenarios, e.g. in the metalstack case.